MachineLearningPrototype/utils/preprocess_data.py

```diff
@@ -15,3 +16,48 @@ def extract_quotation_content(content):
     if (len(substring) >= 1):
         result=' '.join(substring)
         return result
+
+def query_trss_for_jenkins_output(jenkins_url, test_name, config_file_path="config.json"):
+    #Get TRSS servers to query
+    with open(config_file_path) as config_file:
+        config = json.loads(config_file.read())
+        if "trss_servers" not in config:
+            raise Exception("No TRSS servers in config file")
+            return ""
+        servers = config["trss_servers"]
+    
+    #Get build info from TRSS
+    query_url = "https://trss.adoptium.net/api/parseJenkinsUrl"
+    query_params = {"jenkinsUrl": jenkins_url}
+    output = requests.get(query_url, query_params)
+    output_dict = json.loads(output.content)
+
+    if "output" not in output_dict:
+        raise Exception("Invalid jenkins Url")
+        return ""
+    
+    output_dict = output_dict["output"]
+
+    if output_dict["errorMsg"]:
+        raise Exception("Failed to parse Jenkins Url")
```

Comment:
For the exception info, we can contain `output_dict["errorMsg"]` info too.

Reply:
Done! Please take a look. Thank you